### PR TITLE
Add a way to create a commit as a buffer

### PR DESCRIPTION
**Pull request by carlosmn**
This is instead of writing it out to disk.

### CODE REVIEW

File: ext/rugged/rugged_commit.c [hunk 1]
```diff
@@ -662,11 +693,77 @@ static VALUE rb_git_commit_extract_signature(int argc, VALUE *argv, VALUE self)
 	return ret;
 }
 
+/*
+ *  call-seq:
+ *    Commit.create_buffer(repository, data = {}) -> commit
```
Comment (vmg) [inline ext/rugged/rugged_commit.c]: This should say `-> String`. But maybe `create_buffer` is not the ideal name here. `Commit#compose` maybe?

Comment (carlosmn) [inline ext/rugged/rugged_commit.c]: `#compose` seems like a weird name... *we're* not composing anything here... maybe `#create_as_string()` or something with `in_memorry` in the name?

Comment (vmg) [inline ext/rugged/rugged_commit.c]: `#create_to_s` sounds more Ruby-like

Comment (carlosmn) [inline ext/rugged/rugged_commit.c]: Let's do that then.

**Comment by carlosmn** [comment]
Updated with the new name.

**Pull request merged**